### PR TITLE
Use result set wrapper (#68)

### PR DESCRIPTION
**Pull request by sdelatorrep**
This PR changes the general structure of the responses, but keeps fields like `datasetAlleleRespones`, which will be removed/updated in a different PR.

### CODE REVIEW

**Review by mbaudis — DISMISSED** [review]
Looks good to me, for the minimal representation.
Possible/future change could be 
* limiting the "type" values to a predefined selection of terms or objects
* adding a `description` parameter
* adding counts ... (i.e. the dataset count parameters; `variantCount`, `callCount`, `sampleCount`)

=> in principle extend it to a subset of the `BeaconDataset` parameters, w/o making it explicitly a dataset

I'm submitting this as "approval", since what is there looks good, but with the suggestion of expansion, especially regarding the **counts**. We will otherwise map those into "info", but that is a hack.

**Comment by jrambla** [comment]
I'm planning to review the relationship with Resultset, Dataset, Cohorts and counters in the next days, so I hope to have considered answers, comments or suggestions soon.
I see that the current PR is adding one resultset schema per object type... is that really necessary?

**Comment by sdelatorrep** [comment]
@jrambla , I didn't know of any other way to specify different types in `results` field. 
So, you have one base type `ResultSet` with the common properties, and then each response extends this type and adds their own `results` with the correct type:
```yaml
    VariantResultSet:
      allOf:
      - description: |
          TBD
      - $ref: '#/components/schemas/ResultSet'
      - type: object
        properties:
          results:
            type: array
            items:
              $ref: '#/components/schemas/VariantReponseResults'
```

File: beacon.yaml [hunk 1]
```diff
@@ -1705,33 +1751,15 @@ components:
         numTotalResults:
           type: integer
           minimum: 0
-        results:
```
Comment (Tom-Shorter) [inline beacon.yaml:1708]: If I'm reading this correctly then results will disappear and be replaced by resultSet, shouldn't results still be an array with the items being the resultSets?

Comment (sdelatorrep) [reply beacon.yaml:1708]: Good catch! `resultSets` must be an array.

**Comment by mbaudis** [comment]
@sdelatorrep @Tom-Shorter Ouch. Yes, As in the previous discussions results -> resultSets:

```
resultSets:
  - (instance of VariantResultSet ...)
  - (another ...)
 ```

File: beacon.yaml (hunk 1 above)
Comment (mbaudis) [reply beacon.yaml:1708]: Fast :-)

Comment (sdelatorrep) [reply beacon.yaml:1708]: Fixed :)

**Review by mbaudis — DISMISSED** [review]
With fix - again approved, but caveats about recommended per-resultSet count values (i.e. `ResultSet` should be more similar to `BeaconDataset` w/ `callCount` ... etc. - these values can be in principle retrieved separately but are valuable for frequencies etc.).

**Review by mbaudis — APPROVED** [review]

**Review by Tom-Shorter — APPROVED** [review]
Looks good to me!

**Pull request merged**